cordformation/src/main/kotlin/net/corda/plugins/Dockerform.kt

```diff
@@ -201,6 +207,30 @@ open class Dockerform @Inject constructor(objects: ObjectFactory) : Baseform(obj
             services[it.containerName] = service
         }
 
+        if (external.containerImage.isNotEmpty()) {
+            val extra = mutableMapOf(
+                    "image" to external.containerImage,
+                    "ports" to external.servicePorts.toList().map{QuotedString("${it}:${it}")},
+                    "expose" to external.servicePorts
+            )
+            if(external.volumes.isNotEmpty()){
+                extra["volumes"] = external.volumes.toList().map{ "${it["sourceFile"]}:${it["deploymentPath"]}" }
+            }
+            if(external.environment.isNotEmpty()){
+                extra["environment"] = external.environment.toList().map { "${it.first}=${it.second}" }
+            }
+            if(external.containerName.isNotEmpty()){
+                extra["container_name"] = external.containerName
+            }
```

Comment:
`containerName` has a convention of `external-service`, so do you need to check whether it has a value? You seem to be assuming it _always_ has a value here:
```kotlin
services["${external.containerName}"] = extra
```
Shouldn't the expectation be consistent?

Reply:
Yeah, I just stuttered here.